catalog/app/data/workflows.js

```diff
@@ -0,0 +1,70 @@
+import * as R from 'ramda'
+
+import yaml from 'utils/yaml'
+
+/* TODO:
+   ```
+   class Workflows {
+     static parse(rawData) {
+      return new Workflows(parsedData)
+     }
+
+     static nil() { // instead of emptyWorkflowsConfig
+      return new Workflows()
+     }
+   }
+   ```
+*/
+
+export const workflowNotAvaliable = Symbol('not available')
+
+export const workflowNotSelected = Symbol('not selected')
+
+function getNoWorkflow(data, hasConfig) {
+  return {
+    isDefault: !data.default_workflow,
+    slug: hasConfig ? workflowNotSelected : workflowNotAvaliable,
+  }
+}
+
+export const emptyWorkflowsConfig = {
+  isRequired: false,
+  workflows: [getNoWorkflow({}, false)],
+}
+
+function parseSchema(schemaSlug, schemas) {
+  return {
+    url: R.path([schemaSlug, 'url'], schemas),
+  }
+}
+
+function parseWorkflow(workflowSlug, workflow, data) {
+  return {
+    description: workflow.description,
+    isDefault: workflowSlug === data.default_workflow,
+    name: workflow.name,
+    schema: parseSchema(workflow.metadata_schema, data.schemas),
+    slug: workflowSlug,
+  }
+}
+
+export default function parseWorkflows(workflowsYaml) {
```

Comment:
making this export default doesn't make much sense to me, i'd rather keep it a regular named export (e.g. `parse`)

Reply:
It's the main thing for this module. It is the module, and other stuff is just auxiliary.

The same thing like `import React, { useState, useCallback } from 'react'`